add SVG capabilities

thomasleveil opened the pull request:
This is a quick implementation of SVG support.

In it's current state it does produce a SVG file which can at least be displayed in Inkscape. This is my first time playing with SVG, so there must be things to improve in the produced document format.

I hope this is enough to get more SVG savvy people involved.

----

Example of produced svg file 
[signatures.zip](https://github.com/gcacace/android-signaturepad/files/143668/signatures.zip)

thomasleveil commented:
Ideas of svg optimizations: https://jakearchibald.github.io/svgomg/

gcacace commented on SignaturePad-Example/src/main/java/com/github/gcacace/signaturepad/MainActivity.java:
```diff
@@ -106,4 +111,21 @@ public boolean addSignatureToGallery(Bitmap signature) {
         return result;
     }
+
+    public boolean addSignatureToGallery(String signatureSvg) {
```
Please rename this function in something like `addSvgSignatureToGallery`, and rename also the other function generating the JPG/PNG image in `addJpgSignatureToGallery`.

gcacace commented on signature-pad/src/main/java/com/github/gcacace/signaturepad/utils/Svg.java:
```diff
@@ -0,0 +1,28 @@
+package com.github.gcacace.signaturepad.utils;
+
+public class Svg {
```
What do you think about renaming this class in something like `SvgCurve`, then create an instance of it and store inside it the list of `<path/>`, finally create a `build()` method to apply header, footer and return the SVG XML.

gcacace commented on signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java:
```diff
@@ -33,6 +34,8 @@
     private float mLastWidth;
     private RectF mDirtyRect;
 
+    private final StringBuilder mSvgBuilder = new StringBuilder();
```
I would prefer having an instance of `SvgCurve` instead.

gcacace commented on signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java:
```diff
@@ -199,6 +203,12 @@ public boolean isEmpty() {
+    public String getSignatureSvg() {
+        int width = getTransparentSignatureBitmap().getWidth();
+        int height = getTransparentSignatureBitmap().getHeight();
+        return Svg.header(width, height) + mSvgBuilder.toString() + Svg.footer();
```
In that case, as per my previous comments, here you should call `svgCurve.build()` to get the SVG XML string.

thomasleveil commented:
Thank you the code review and valuable feedbacks :memo:

last commit introduces a new `SvgBuilder` class in which all SVG related code is enclosed. This is the place where further optimizations on the produced SVG document can take place without fearing regressions on the rest of the project.

Please let me know if you want me to squash those commits

gcacace commented:
Thanks for your commits! If you can, please squash all in just a single commit so I can merge with master.

thomasleveil commented:
squashed. I left out the first commit which has nothing to do with SVG and added two commit to illustrate how one can contribute by optimizing the produced SVG document.
